index.js

```diff
@@ -9,12 +9,14 @@ module.exports = {
   name: 'emberfire',
 
   included: function included(app) {
-    this._super.included(app);
+    this._super.included.apply(this, arguments);
+    var app;
 
     // make sure app is correctly assigned when being used as a nested addon
-    if (app.app) {
-      app = app.app;
-    }
+    var current = this;
+    do{
+      app = current.app || app;
+    } while (current.parent.parent && (current = current.parent));
```

Comment:
This looks a bit of magic, without the PR description I would not know what this loop is doing. Perhaps we should create a method to lookup the app. Also, if `current.parent` is undefined, this will throw a TypeError, we should safe check that.

Reply:
Yes you are right, this was a patch for the issue. I hope some one in the core team would look at this and have the time to implement a proper solution, in the mean time I'm patching with this all my projects.